### PR TITLE
Add two slides on CMake

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Fixes: #123

### CODE REVIEW

File: talk/tools/compiling.tex [hunk 1]
```diff
@@ -109,6 +109,48 @@
   \end{minted}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitle{CMake}
+  \begin{block}{}
+    \begin{itemize}
+      \item a cross platform meta build system
+      \item generates platform specific build systems
```
Comment (hageboeck) [inline talk/tools/compiling.tex:117]:
```suggestion
      \item a cross-platform meta build system
      \item generates platform-specific build systems
```

File: talk/tools/compiling.tex [hunk 2]
```diff
@@ -109,6 +109,48 @@
...
+    \end{minted}
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{CMake - Building}
+  \begin{block}{Building a CMake-based project}
+    Start in the directory with the top-level \texttt{CMakeLists.txt}:
+    \begin{minted}[linenos=true,autogobble]{bash}
+      mkdir build # will contain all build related files
```
Comment (hageboeck) [inline talk/tools/compiling.tex:140]:
```suggestion
      mkdir build # will contain all build-related files
```

File: talk/tools/compiling.tex [hunk 3]
```diff
@@ -109,6 +109,48 @@
...
+      cd build
+      cmake ..    # configures and generates a build system
+      cmake -DCMAKE_BUILD_TYPE=Release .. # pass arguments
+      ccmake .    # change configuration and regenerate
```
Comment (hageboeck) [inline talk/tools/compiling.tex:144]:
```suggestion
      ccmake .    # change configuration using a GUI
```

File: talk/tools/compiling.tex [hunk 4]
```diff
@@ -109,6 +109,49 @@
...
+    \end{minted}
+  \end{block}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitle{CMake - Building}
+  \begin{block}{Building a CMake-based project}
+    Start in the directory with the top-level \texttt{CMakeLists.txt}:
+    \begin{minted}[linenos=true,autogobble]{bash}
+      mkdir build # will contain all build-related files
```
Comment (bernhardmgruber) [inline talk/tools/compiling.tex:140]: That is where I usually create build folders (top level inside the working copy), but I have also seen people create the build folder completely outside the working copy. What's your take on this?

Comment (hageboeck) [reply talk/tools/compiling.tex:140]: I do both. Imagine the case where you have code on a network file system and object files on an ssd, space problems on one device, stuff you want to throw away in `/tmp`, tons of things.
I think it's fine to show build folder in source folder, but we should *say* in the lecture that you can put it anywhere.

**Pull request merged**